config/src/config/transaction_filter_type.rs

```diff
@@ -0,0 +1,192 @@
+// Copyright © Aptos Foundation
+// Parts of the project are originally copyright © Meta Platforms, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use aptos_crypto::HashValue;
+use aptos_types::{
+    account_address::AccountAddress,
+    transaction::{SignedTransaction, TransactionPayload},
+};
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+enum Matcher {
+    All,
+    BlockId(HashValue),
+    TransactionId(HashValue),
+    Sender(AccountAddress),
+    ModuleAddress(AccountAddress),
```

Comment:
Silly question: Do we want the ability to disable transaction payload types? e.g., disable all transactions of type `TransactionPayload::Script(...)`, `TransactionPayload::ModuleBundle(...)` or `TransactionPayload::Multisig(...)`?

The move experts will probably have to confirm/deny, but I'm just trying to think about edge-cases that may be exploitable through those transaction types and if a filter will be useful.

'cc @davidiw @movekevin, any thoughts? 😄

Reply:
We can extend this framework further to allow more filters..if needed, may be we can do that as a followup?